aidial_adapter_vertexai/chat/gemini/adapter.py

```diff
@@ -196,12 +200,17 @@ async def process_chunks(
                     yield content
 
                 await create_function_calls(candidate, consumer, tools)
-                await create_grounding(candidate, consumer)
+                is_grounding_added = await create_grounding(candidate, consumer)
                 await create_attachments_from_citations(candidate, consumer)
                 await set_finish_reason(candidate, consumer)
 
             if chunk.usage_metadata:
-                await set_usage(chunk.usage_metadata, consumer)
+                await set_usage(
```

Comment:
Could it be that the chunk with `usage_metadata` and the chunk with the grounding are two **different** chunks?

Reply:
Fixed